modules/build/src/main/scala/scala/build/Inputs.scala

```diff
@@ -233,6 +237,18 @@ object Inputs {
       .sortBy(_.subPath.segments)
   }
 
+  def projectSettingsFiles(elements: Seq[Inputs.Element]): Seq[Inputs.ProjectSettings] =
+    elements.flatMap {
+      case f: ProjectSettings => Seq(f)
+      case d: Directory       => Inputs.configFileFromDirectory(d)
+      case _                  => Nil
+    }.distinct
+
+  def configFileFromDirectory(d: Inputs.Directory): Seq[Inputs.ProjectSettings] =
+    if (os.exists(d.path / "project.settings.scala"))
+      Seq(Inputs.ProjectSettings(d.path, os.SubPath("project.settings.scala")))
```

Comment:
NIT I believe this works (and is easier to read):
```suggestion
      Seq(Inputs.ProjectSettings(d.path, os.sub / "project.settings.scala"))
```